apps/src/code-studio/components/progress/StageProgress.story.jsx

```diff
@@ -6,99 +6,98 @@ import stageLock from '../../stageLockRedux';
 import progress, { initProgress } from '../../progressRedux';
 
 const activityPuzzle = {
-  ids: [
-    123
-  ],
   activeId: 123,
   position: 1,
-  kind: "puzzle",
-  icon: "",
-  title: 1,
   url: "http://studio.code.org/s/course1/stage/3/puzzle/2",
-  freePlay: false,
-  is_concept_level: false,
+  levels: [{
+    id: 123,
+    kind: "puzzle",
+    icon: "",
+    title: 1,
+    freePlay: false,
+    is_concept_level: false,
+  }],
 };
 
 const conceptPuzzle = {
-  ids: [
-    5086
-  ],
   activeId: 5086,
   position: 2,
-  kind: "puzzle",
-  icon: "fa-file-text",
-  title: 2,
   url: "http://studio.code.org/s/csp1/stage/2/puzzle/1",
-  freePlay: false,
   progression: "Lesson Vocabulary & Resources",
-  is_concept_level: true,
+  levels: [{
+    id: 5086,
+    kind: "puzzle",
+    icon: "fa-file-text",
+    title: 2,
+    freePlay: false,
+    is_concept_level: true,
+  }],
 };
 
 const assessment1 = {
-  ids: [
-    2441
-  ],
   activeId: 2441,
   position: 3,
-  kind: "assessment",
-  icon: "fa-check-square-o",
-  title: 3,
   url: "http://studio.code.org/s/csp1/stage/2/puzzle/3",
-  freePlay: false,
-  progression: "Check Your Understanding"
+  progression: "Check Your Understanding",
+  levels: [{
+    id: 2441,
+    kind: "assessment",
+    icon: "fa-check-square-o",
+    title: 3,
+    freePlay: false,
+  }],
 };
 
 const assessment2 = {
-  ids: [
-    2444
-  ],
   activeId: 2444,
   position: 4,
-  kind: "assessment",
-  icon: "fa-check-square-o",
-  title: 4,
   url: "http://studio.code.org/s/csp1/stage/2/puzzle/4",
-  freePlay: false,
-  progression: "Check Your Understanding"
+  progression: "Check Your Understanding",
+  levels: [{
+    id: 2444,
+    kind: "assessment",
+    icon: "fa-check-square-o",
+    title: 4,
+    freePlay: false,
+  }]
 };
 
 const assessment3 = {
-  ids: [
-    2744
-  ],
   activeId: 2744,
   position: 5,
-  kind: "assessment",
-  icon: "fa-check-square-o",
-  title: 5,
   url: "http://studio.code.org/s/csp1/stage/2/puzzle/5",
-  freePlay: false,
-  progression: "Check Your Understanding"
+  progression: "Check Your Understanding",
+  levels: [{
+    id: 2744,
+    kind: "assessment",
+    icon: "fa-check-square-o",
+    title: 5,
+    freePlay: false,
+  }],
 };
 
 const unplugged = {
-  ids: [
-    2093
-  ],
   activeId: 2093,
-  is_concept_level: false,
-  kind: 'unplugged',
-  // kind: 'puzzle',
   position: 1,
-  title: 1,
   url: "http://studio.code.org/s/course1/stage/1/puzzle/1",
+  levels: [{
+    id: 2093,
+    is_concept_level: false,
+    kind: 'unplugged',
+    title: 1,
+  }]
 };
 
 export default storybook => {
-  const createStoreForLevels = (levels, currentLevelIndex) => {
+  const createStoreForScriptLevels = (script_levels, currentLevelIndex) => {
```

Comment:
I find myself that some places in our JS we seem to expect `script_levels` and some places we use `scriptLevels`. Really it should probably be `scriptLevels` everywhere. The right way to do this is probably convert it at the point where we stick it into redux. I think that means handling this in `processedStages`